docs/specs/field_edit_markup.md

```diff
@@ -0,0 +1,494 @@
+# Field edit markup
+
+## Common aspects
+
+### Wrapped by a custom element
+
+As described in [EZP-27576](https://jira.ez.no/browse/EZP-27576), when used
+inside Hybrid Platform UI, the edit markup for a given field should be wrapped
+by a custom element. By default, this custom element is `ez-field-edit` and it
+expects `fieldtype` attribute to be filled with the Field Type identifier.
+
+For a Content Type with a Text Line, a Map Location and a Text Block fields, the
+Content edit form would look like:
+
+```html
+<form>
+  <ez-field-edit fieldtype="ezstring">
+     <!-- markup for Text Line -->
+  </ez-field-edit>
+  <ez-field-edit-maplocation fieldtype="ezgmaplocation">
+     <!-- markup for Map Location -->
+  </ez-field-edit>
+  <ez-field-edit fieldtype="eztext">
+     <!-- markup for Text Block -->
+  </ez-field-edit>
+</form>
+```
+
+### Base structure
+
+The markup for a given field is always wrapped in a `div` with at least 2
+classes:
+
+* `ez-field-edit`
+* `ez-field-edit-<fieldtypeidentifier>`
+
+If the Field is marked as required, the class `ez-field-edit-required` is added
+as well.
+
+If the Field is not translatable and the user is translating the Content item
+the class `ez-field-edit-disabled` is added.
+
+The `div` starts with a `div` with the class `ez-field-edit-text-zone`. This
+`div` contains the Field Definition name wrapped in a `label` element. The
+`for` attribute value of this label is filled with the `id` of an input
+generated for the Field Type, usually the first one with some exceptions.
+
+Then the specific to the Field Type part is wrapped in a `div` with the class
+`ez-field-edit-ui`.
+
+If the Field has a description, this description is added in a paragraph as the
+last element of the `div.ez-field-edit`. If the Field is not translatable and
+the user is translating a Content, the description is replaced with the non
+translatable notice.
+
+So for a required Field with a description, the base structure is:
+
+```html
+<div class="ez-field-edit ez-field-edit-ezstring ez-field-edit-required">
+    <div class="ez-field-edit-text-zone">
```

Comment:
self note: this is also where validation messages appear. Open question: how to share message between server and frontend?